python/pyarrow/includes/libarrow_dataset.pxd

```diff
@@ -314,6 +314,10 @@ cdef extern from "arrow/dataset/api.h" namespace "arrow::dataset" nogil:
             const CExpression& partition_expression,
             CRecordBatchProjector* projector)
 
+    cdef CResult[unordered_map[c_string, shared_ptr[CScalar]]] \
+        CGetPartitionKeys "arrow::dataset::KeyValuePartitioning::GetKeys"(
```

Comment:
One question for my education: `GetKeys` is a method on the `KeyValuePartitioning` class, but since that method doesn't use any of the initialized variabled of the object, you can just use it as a free function like this independent from a KeyValuePartitioning object in C++? (like a static/class method in Python, but then without needing to mark it as such)

Reply:
Yes, as far as cython is concerned a static method can be treated as a free function. `CSetPartitionKeysInProjector` (decl above this one) is another example. I think it works the other way, too: a c++ free function can be exposed as a `@staticmethod` of a cppclass (but haven't got a standing example and I'm not sure why we'd ever need that)